src/jit/importer.cpp

```diff
@@ -7664,6 +7676,21 @@ var_types Compiler::impImportCall(OPCODE                  opcode,
             }
             else
             {
+                if (fatPointerCandidate)
+                {
+                    assert(!bIntrinsicImported);
+                    assert(IsTargetAbi(CORINFO_CORERT_ABI));
+                    if (call->OperGet() != GT_LCL_VAR) // can be already converted by impFixupCallStructReturn.
+                    {
+                        unsigned   calliSlot  = lvaGrabTemp(true DEBUGARG("calli"));
+                        LclVarDsc* varDsc     = &lvaTable[calliSlot];
+                        varDsc->lvVerTypeInfo = tiRetVal;
+                        impAssignTempGen(calliSlot, call, clsHnd, (unsigned)CHECK_SPILL_NONE);
```

Comment:
Why fatpointer calli's need to be transformed to be of the form 
var = call
?

Please add a comment explaining the same here.